contract/models/account_analytic_contract.py

```diff
@@ -73,11 +79,22 @@ class AccountAnalyticContract(models.Model):
         default=lambda self: self.env.user.company_id,
     )
 
+    @api.onchange('contract_type')
+    def _onchange_contract_type(self):
```

Comment:
Is this a real case? You can't modify the contract type through UI, so no onchange is going to be called.

Reply:
In the contract template form view, you can change the contract_type.

Reply:
I think the field should be always invisible and not changeable by UI, as you have 2 separate menus already, and imagine who weird would be to have a sale contract (Maybe a better name should be "Customer Contract"?), generate a customer invoice, and then change to purchase the contract...

Reply:
There are three views:

- the sale contract, where contract_type is invisible.
- the purchase contract, where contract_type is invisible.
- the contract template, where contract_type is visible and changeable.

Reply:
OK, I see. Thanks for clarifying me this.